WordPress/Classes/ViewRelated/Themes/ThemeBrowserViewController.swift

```diff
@@ -0,0 +1,201 @@
+import Foundation
+
+public class ThemeBrowserViewController : UICollectionViewController, UICollectionViewDelegateFlowLayout, NSFetchedResultsControllerDelegate, UISearchBarDelegate {
+    
+    // MARK: - Properties: must be set by parent
+    
+    /**
+     *  @brief      The blog this VC will work with.
+     *  @details    Must be set by the creator of this VC.
+     */
+    private var blog : Blog!
+    
+    // MARK: - Properties: managed object context & services
+    
+    /**
+     *  @brief      The managed object context this VC will use for it's operations.
+     */
+    private let managedObjectContext : NSManagedObjectContext = {
+        ContextManager.sharedInstance()!.newDerivedContext()
```

Comment:
You'd probably want to use the Main Context on this one. Reason is: 

- Derived context's use a private GCD queue
- Since you're attaching a `NSFetchedResultsController` instance to it, you'd need to use the main thread instead (due to UIKit).